nmrfx-structure/src/main/java/org/nmrfx/structure/chemistry/energy/RNARotamer.java

```diff
@@ -772,6 +760,10 @@ public static void setDihedrals(Residue residue, String suiteName) {
      * refinement
      */
     public static void setDihedrals(Residue residue, String suiteName, double sdev, boolean doFreeze) {
+        if (residue == null) {
+            log.warn("Residue is null. Unable to set Dihedrals.");
+            return;
+        }
```

Comment:
Returned early from this method if residue was null, is this okay? Would it be better to throw and illegal argument exception here if residue is null? This method is only used in refine.py

Reply:
Changed to throw and IllegalArgumentException